### PR TITLE
Add changelog for 2022.4.0

### LINKED ISSUE #331: Add changelog to docs
We should have a changelog in the docs so users can better track changes between releases.

### PR DESCRIPTION
**Pull request by consideRatio**
I've tracked some pre-release work in #543, but at this point, we are ready for a proper `2022.4.0` release!

Closes #331. Note that I've added `myst_parser` as a dependency to allow this file to be written in markdown and integrate with the sphinx based website. It renders like this:

image

## About recent self-merge sprint

Please note that I sprinted a bit to get to this point towards the end by self-merging CI related PRs listed below. I hope you found that acceptable. Given that we are short on maintainers to review and there were a lot of sequentials PRs that I perceived had to be made I worried that if I didn't went for this I'd run out of steam to get this release out before I had to shift my focus towards other open source projects and work tasks.
- #530
- #531
- #534
- #536
- #538
- #540
- #545
- #548
- #549
- #550
- #551
- #552
- #554
- #555
- #556

### CODE REVIEW

**Comment by martindurant** [comment]
Hurray!

File: docs/source/changelog.md [hunk 1]
```diff
@@ -0,0 +1,140 @@
...
+The project now adopts [CalVer](https://calver.org/) versioning with a
+`YYYY.MM.MICRO` format like many other Dask organization projects.
+
+### Breaking changes
+
+- `dask-gateway` and `dask-gateway-server` now requires Python 3.8+
+- Breaking changes to the `dask-gateway` Helm chart:
+  - _ACTION REQUIRED_: When upgrading to this version you must also update the Helm chart's bundled CRD resources like this:
+    ```shell
+    kubectl apply -f https://raw.githubusercontent.com/dask/dask-gateway/2022.4.0/resources/helm/dask-gateway/crds/daskclusters.yaml
```
[the remark this answers is not in the document]
Comment (consideRatio) [reply docs/source/changelog.md:18]: @martindurant yepp one strategy to upgrade is to delete all the old ones, then do `helm upgrade` again. Like that, `helm` will find no CRDs are around and dare to install new CRDs. So, deleting them was a trick to get `helm` to install the new ones. If `helm` finds existing CRDs, outdated or not, it doesn't do anything.

This option is to instead add new/modify existing CRDs which also works with less trouble and less possible disruption.

File: docs/source/changelog.md [hunk 2]
```diff
@@ -0,0 +1,140 @@
...
...o%3Adask%2Fdask-gateway+involves%3Apre-commit-ci+updated%3A2020-11-04..2022-04-20&type=Issues) | [@quasiben](https://github.com/search?q=repo%3Adask%2Fdask-gateway+involves%3Aquasiben+updated%3A2020-11-04..2022-04-20&type=Issues) | [@raybellwaves](https://github.com/search?q=repo%3Adask%2Fdask-gateway+involves%3Araybellwaves+updated%3A2020-11-04..2022-04-20&type=Issues) | [@rigzba21](https://github.com/search?q=repo%3Adask%2Fdask-gateway+involves%3Arigzba21+updated%3A2020-11-04..2022-04-20&type=Issues) | [@rileyhun](https://github.com/search?q=repo%3Adask%2Fdask-gateway+involves%3Arileyhun+updated%3A2020-11-04..2022-04-20&type=Issues) | [@rsignell-usgs](https://github.com/search?q=repo%3Adask%2Fdask-gateway+involves%3Arsignell-usgs+updated%3A2020-11-04..2022-04-20&type=Issues) | [@TomAugspurger](https://github.com/search?q=repo%3Adask%2Fdask-gateway+involves%3ATomAugspurger+updated%3A2020-11-04..2022-04-20&type=Issues) | [@wdhowe](https://github.com/search?q=repo%3Adask%2Fdask-gateway+involves%3Awdhowe+updated%3A2020-11-04..2022-04-20&type=Issues) | [@yuvipanda](https://github.com/search?q=repo%3Adask%2Fdask-gateway+involves%3Ayuvipanda+updated%3A2020-11-04..2022-04-20&type=Issues)
```
[the remark this answers is not in the document]
Comment (consideRatio) [reply docs/source/changelog.md:140]: I've used [`github-activity`](https://github.com/executablebooks/github-activity#readme) to generate this list without making any change to the formatting. It looks like this:

image

**Comment by consideRatio** [comment]
> A quick sanity check should be done that the change rst->md didn't break anything.

Everything seems functioning as before, you can inspect the docs by checking out the branch and doing `pip install docs/requirements.txt; cd docs; make devenv`. If you have troubles with pip install requirements `export DASK_GATEWAY_SERVER__NO_PROXY=true` first to avoid compiling a golang binary.

**Comment by consideRatio** [comment]
I've now made an end to end test of the 2022.4.0-beta.1 release made, and everything works! I'll go for a merge and release of this to start working on what we need to get done next in conda-forge.

**Comment by consideRatio** [comment]
I've cut a release by bumping some versions according to RELEASE.md in a commit, and pushed a tag.

The result is that the following workflows triggered and succeed:
- https://github.com/dask/dask-gateway/runs/6116641695
- https://github.com/dask/dask-gateway/runs/6116626886

And we now have PyPI artifacts, a Helm chart and two images published:
- https://pypi.org/project/dask-gateway/2022.4.0/#files
- https://pypi.org/project/dask-gateway-server/2022.4.0/#files
- https://helm.dask.org
  image
- https://github.com/dask/dask-gateway/pkgs/container/dask-gateway
- https://github.com/dask/dask-gateway/pkgs/container/dask-gateway-server

**Comment by choldgraf** [comment]
This is amazing y'all 🙂

**Pull request merged**